File: Website/InfoTerraTerra/InfoTerraTerra_Library/Newsletter/NewsletterIscrittiStatistics.cs
namespace InfoTerraTerra_Library.Newsletter;

public class NewsletterIscrittiStatistics
{
    public string[] AllNewsletterEmail { get; }
    public string? LastIscrittoDate { get; }
    public KeyValuePair<string, string>[] Counters { get; }

    public NewsletterIscrittiStatistics(
        string[] allNewsletterEmail,
        string? lastIscrittoDate)
    {
        AllNewsletterEmail = allNewsletterEmail;
        LastIscrittoDate = lastIscrittoDate;
        
        Counters = new[]
        {
            new KeyValuePair<string, string>("Numero iscritti alla newsletter", GetNumeroIscrittiText()),
            new KeyValuePair<string, string>("Data ultimo iscritto alla newsletter", LastIscrittoDate ?? "😟"),
        };
    }
    
    private string GetNumeroIscrittiText()
    {
        return AllNewsletterEmail.Length switch
        {
            0 => "Nessuno",
            1 => "Uno solo 🧐",
            _ => $"{AllNewsletterEmail.Length} iscritti"
        };
    }
}

File: dataLayer/newsletter/NewsletterIscrittiStatistics.ts
import {KeyValuePair} from "../../model/KeyValuePair";

export class NewsletterIscrittiStatistics {
    public allNewsletterEmail: string[];
    public counters: KeyValuePair<string, string>[];

    constructor(allNewsletterEmail: string[]) {
        this.allNewsletterEmail = allNewsletterEmail;

        this.counters = [
            {
                key: "Numero iscritti alla newsletter",
                value: this.getNumeroIscrittiText()
            }
        ];
    }

    private getNumeroIscrittiText(): string {
        switch (this.allNewsletterEmail.length) {
            case 0:
                return "Nessuno";
            case 1:
                return "Uno solo 🧐";
            default:
                return `${this.allNewsletterEmail.length} iscritti`;
        }
    }
}
